### PR TITLE
Replace MeSH tree numbers with IDs

### LINKED ISSUE #698: Some MeSH xrefs point to tree locations rather than unique IDs
I have two comments regarding MeSH cross-references:
1. Currently, MeSH xrefs are in terms of tree numbers. For example, the obo file contains `xref: MESH:A01.456.505.733` rather than `xref: MESH:D009666`. A tree number represents a path up the MeSH hierarchy from term to top-level category. Therefore, MeSH terms can have multiple tree numbers, and tree numbers are subject to change whenever the hierarchy is reorganized, even if the term remains. Is there a reason tree numbers are used instead of unique ids? If you would like to switch, you may find my mapping helpful ([notebook](https://github.com/dhimmel/uberon/blob/7de0ed6238c26ea82accae6fc57accd1b845111d/mesh-map.ipynb), [tsv file](https://github.com/dhimmel/uberon/blob/7de0ed6238c26ea82accae6fc57accd1b845111d/data/mesh-map.tsv)).
2. I noticed MeSH mappings only exist for UBERON terms but not terms populated from CL. Since many CL terms contain MeSH terms as citations for the definition (example: `def: "A male germ cell that develops from the haploid secondary spermatocytes. Without further division, spermatids undergo structural changes and give rise to spermatozoa." [MESH:A05.360.490.890.860]`), it appears some of the hard work has already been done. Why are definition sources not included as xrefs and is this the right location to submit CL feature requests?

### PR DESCRIPTION
**Pull request by bgyori**
This PR replaces a number of MeSH tree numbers appearing as xrefs with proper MeSH entry IDs.

Currently, UBERON provides 923 xrefs to MESH out of which 643 are valid IDs and 280 are tree numbers.

A total of 4 such tree numbers are valid and can be replaced by a corresponding MeSH ID using simple lookup. The remaining 276 tree numbers are deprecated and cannot simply be looked up in the current MeSH release.

Therefore I used Gilda (https://github.com/indralab/gilda) to automatically map UBERON entries that use invalid MeSH tree numbers as xrefs to MeSH IDs, using lexical matches. This yielded 201 further mappings.

A total of 75 invalid MeSH tree numbers cannot be mapped with either of these methods and remain invalid. However, these appear to be still "mappable" using manual lookups.

Separate from this PR, I also created an overview table of the invalid MeSH xrefs and the mappings proposed in this PR: https://github.com/bgyori/uberon/blob/mesh_review/uberon_mesh_mappings.csv. This will hopefully help review these changes.

Partially fixes #698.

### CODE REVIEW

**Comment by bgyori** [comment]
I found one example of a MeSH tree number xref that is currently a valid MeSH tree number but the mapping is not valid anymore (presumably due to MeSH having changed since the mapping was created): https://github.com/bgyori/uberon/blob/mesh_review/uberon_mesh_mappings.csv#L76
This implies that even the tree-number-based mappings need to be reviewed and potentially manually modified.

**Review by dosumis — APPROVED** [review]
This looks great.  Really nice to be able to review in table form too. 
Even the potentially dodgy mapping you mapped is actually v.close - may actually be correct https://meshb.nlm.nih.gov/record/ui?ui=D034941 (It looks like there is not universal agreement about what to include in the arm - see notes on https://www.ebi.ac.uk/ols/ontologies/uberon/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0001460&viewMode=PreferredRoots&siblings=false)

@rays22 could you review too - just  eyeball the mappings (see link to table posted by @bgyori) and check any others that look odd.

File: src/ontology/uberon-edit.obo [hunk 1]
```diff
@@ -22542,7 +22542,6 @@ xref: EHDAA:8275
 xref: EMAPA:17413
 xref: GAID:52
 xref: MA:0000033
-xref: MESH:A01.378.800
 xref: MESH:D001132
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:22545]: This looks correct.

**Review by rays22 — APPROVED** [review]
Looks OK.

**Comment by dhimmel** [comment]
> I found one example of a MeSH tree number xref that is currently a valid MeSH tree number but the mapping is not valid anymore (presumably due to MeSH having changed since the mapping was created). permalink at [`uberon_mesh_mappings.csv#L76`](https://github.com/bgyori/uberon/blob/3f3fe53c6b8d3ed46e93d783962dda614ab6f7f4/uberon_mesh_mappings.csv#L76)

So you're saying that `UBERON:0001460` / arm is not an exact match for `mesh:D034941` / Upper Extremity? I agree. It looks like there is now [`mesh:D001132`](https://id.nlm.nih.gov/mesh/D001132.html) which is an exact match for arm.

**Comment by bgyori** [comment]
Yes, that's what I meant. In the final PR, the correct  D001132 xref is used.

**Pull request merged**